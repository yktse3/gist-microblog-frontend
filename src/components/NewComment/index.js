import React, { Component } from 'react';
import {
  func,
  string,
} from 'prop-types';
import {
  CommentContent,
} from './styles';
import {
  SubmitBtn,
} from '../Editor/styles';
import {
  Card,
} from '../Article/styles';

class Editor extends Component {
  static propTypes = {
    onSubmit: func.isRequired,
    articleID: string.isRequired,
  };

  state = {
    content: '',
  };

  onChange = (e) => {
    this.setState({
      content: e.target.value,
    });
  }

  onSubmit = () => {
    this.setState({
      content: '',
    });
    this.props.onSubmit(this.props.articleID, this.state.content);
  }

  render() {
    return (
      <Card>
        <CommentContent
          name="content"
          placeholder="Comment"
          value={this.state.content}
          onChange={this.onChange}
          autoComplete="off"
        />
        <SubmitBtn
          onClick={this.onSubmit}
        >
        Submit
        </SubmitBtn>
      </Card>
    );
  }
}

export default Editor;
