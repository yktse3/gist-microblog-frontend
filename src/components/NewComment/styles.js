import styled from 'styled-components';
import {
  Content,
} from '../Editor/styles';

export const CommentContent = styled(Content)`
  height: 100px;
`;