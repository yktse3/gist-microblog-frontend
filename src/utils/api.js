const CLIENT_ID = 'b77a50cf0d2a3029972c';
const CLIENT_SECRET = 'f29284e62d8c8bdd0fba0b304b7c5c243eec0d88';
const HEADER_ACCEPT_JSON = 'application/json';
const HEADER_ACCEPT_GITHUB_V3 = 'application/vnd.github.v3+json';

const defaultConfig = {
  accessToken: null,
};

class API {
  constructor(config = defaultConfig) {
    this._config = config;
  }

  changeAccessToken = (accessToken) => {
    this._config.accessToken = accessToken;
  }

  getAccessToken = async ({ code }) => {
    const url = 'http://localhost:3100/login';
    const body = {
      client_id: CLIENT_ID,
      client_secret: CLIENT_SECRET,
      code,
    };

    const config = {
      method: 'POST',
      headers: {
        Accept: HEADER_ACCEPT_JSON,
        'Content-Type': HEADER_ACCEPT_JSON,
        'Access-Control-Allow-Origin': '*',
      },
      body: JSON.stringify(body),
    };

    try {
      const result = await fetch(url, config);
      const response = await result.json();
      return response;
    } catch (e) {
      return Promise.reject(e);
    }
  };
}

export default new API();
