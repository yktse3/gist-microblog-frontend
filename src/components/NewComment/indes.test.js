import React from 'react';
import renderer from 'react-test-renderer';
import { noop } from 'utils/helpers';
import { mount } from 'enzyme';
import NewComment from './index';

describe('NewComment', () => {
  it('render successfully when enabled', () => {
    const component = renderer.create(
      <NewComment
        onSubmit={noop}
        articleID="articleID"
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('change state when typing in comment box', () => {
    const wrapper = mount(
      <NewComment
        onSubmit={noop}
        articleID="articleID"
      />,
    );

    wrapper
      .find('textarea')
      .first()
      .simulate('change', {
        target: {
          value: 'new comment',
        },
      });
    expect(wrapper.state('content')).toEqual('new comment');
  });

  it('calls onSubmit with correct parameters', () => {
    const onSubmit = jest.fn();
    const wrapper = mount(
      <NewComment
        onSubmit={onSubmit}
        articleID="articleID"
      />,
    );

    wrapper.setState({
      content: 'new comment',
    });

    wrapper
      .find('button')
      .first()
      .simulate('click');

    expect(onSubmit).toBeCalledWith('articleID', 'new comment');

    expect(wrapper.state()).toEqual({
      content: '',
    });
  });
});
