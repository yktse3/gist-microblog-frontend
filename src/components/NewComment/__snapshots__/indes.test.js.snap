// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewComment render successfully when enabled 1`] = `
<div
  className="sc-bZQynM bTIDzm"
>
  <textarea
    autoComplete="off"
    className="sc-EHOje fwxGOl sc-bwzfXH jGzomT"
    name="content"
    onChange={[Function]}
    placeholder="Comment"
    value=""
  />
  <button
    className="sc-htpNat eTyvdL"
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
